src/api/store/graph.py

```diff
@@ -87,6 +87,55 @@ def graph_actions_completed(self, context: Context, args) -> (Graph, MassEnergiz
       return None, CustomMassenergizeError(e)
 
 
+  def graph_actions_completed_by_team(self, context: Context, args) -> (Graph, MassEnergizeAPIError):
+    try:
+
+      team_id = args.get('team_id', None)
+
+      if not team_id:
+        return None, CustomMassenergizeError("Missing team_id field")
+
+      team: Team = Team.objects.get(id=team_id)
+
+      if not team:
+        return None, InvalidResourceError()
+
+      if context.is_prod and not context.user_is_admin():
+        if not team.is_published:
+          return None, CustomMassenergizeError("Content Not Available Yet")
+  
+      members = TeamMember.objects.filter(team=team).all()
+
+      completed_action_rels = []
+      for member in members:
+        completed_action_rels.extend(member.user.useractionrel_set.filter(status="DONE").all())
+
+      categories = TagCollection.objects.get(name="Category").tag_set.order_by("name").all()
+
+      prefetch_related_objects(completed_action_rels, "action__tags")
+      data = [
+        {
+          "id"   : category.id,
+          "name" : category.name,
+          "value": len(list(filter(lambda action_rel : category in action_rel.action.tags.all(), completed_action_rels)))
```

Comment:
i love the mastery and clean use of list comprehension here.  However i would go for more code comprehension here just for easier maintenance.  I can understand what you have done, but as we get more and more people to help out it might be better to break complex pieces like this into smaller chunks just for easier code readability.

Reply:
Also i think you dont need to do `list(` when you are using filter since it already returns an iterable.  so `len(filter...` would be more efficient.

Reply:
removed `all()` and made changes for readability. but as for removing the `list` cast, it causes `TypeError: object of type 'filter' has no len()`. While `filter` returns an interable, `len` takes in a sequence or collection, specifically.